content/en/docs/concepts/services-networking/service.md

```diff
@@ -619,6 +619,77 @@ You can integrate with [Gateway](https://gateway-api.sigs.k8s.io/) rather than S
+balancing targets _as long as_ the node is not being deleted and Kube-proxy is
+healthy. In this mode: Kubernetes configures the load balancer health check to
+target the service proxy's readiness port and path. In the case of Kube-proxy
+this evaluates to: `${NODE_IP}:10256/healthz`. Kube-proxy will return either an
+HTTP code 200 or 503. It will return 200 if a) Kube-proxy is healthy, meaning:
+it's able to progress programming the network and isn't timing out while doing
+so (the timeout is defined to be: `2 *
+kubeProxyIPTablesConfiguration.syncPeriod`) and b) the Node is not being
+deleted. The reason why Kube-proxy returns 503 and marks the node as not
+eligible when it's being deleted, is because Kube-proxy supports connection
+draining for terminating nodes. A couple important things occur from the point
+of view of a Kubernetes load balancer when a node _is being_ / _is_ deleted.
+
+While deleting:
+
+* Kube-proxy will start failing its readiness probe and essentially mark the
+   node as not eligible for load balancer traffic. The load balancer health
+   check failing causes load balancers which support connection draining to
+   allow existing connections to terminate, and block new connections from
+   establishing.
+
+When deleted:
+
+* The service controller in the Kubernetes cloud controller manager removes the
+  node from the referenced set of eligible targets. Removing any instance from
+  the load balancer's set of backend targets immediately terminates all
+  connections. This is also the reason Kube-proxy first fails the health check
+  while the node is deleting.
+
+It's important to note for Kubernetes vendors that if any vendor configures the
+Kube-proxy readiness probe as a liveness probe: that Kube-proxy will start
+restarting continuously when a node is deleting until it has been fully deleted.
+
+Users deploying Kube-proxy can inspect both the readiness / liveness state by
+evaluating the metrics: `proxy_livez_total` / `proxy_healthz_total`. Both
+metrics publish two series, one with the 200 label and one with the 503 one.
+
+For services of `externalTrafficPolicy: local`:  Kube-proxy will return 200 if
+a) Kube-proxy is healthy/ready and b) has a local endpoint on the node in
+question. Note: node deletion does not have an impact on Kube-proxy's return
+code for what concerns load balancer health checks. The reason for this is:
+deleting nodes could end up causing an ingress outage should all endpoints
+simultaneously be running on said nodes.
+
+It's important to note that the configuration of load balancer health checks is
+specific to each cloud provider, meaning: different cloud providers configure
+the health check in different ways. The three main cloud providers do so in the
+following way:
+
+* GCP: probes port 10256 (Kube-proxy's healthz port)
+* AWS: if ELB; probes the first NodePort defined on the service spec
+* Azure: probes all NodePort defined on the service spec.
```

Comment:
- It's not the cloud providers that configure health checks; it's the _integration code_ that does the configuring.
- Please list these firms in alphabetical order.

Reply:
What is the "integration code", is that what we call everything which lives here: https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/cloud-provider ?

Reply:
> What is the "integration code", is that what we call everything which lives here: https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/cloud-provider ?

Not exactly. I think the implementation of code to configure health checking lives within repos like https://github.com/kubernetes-sigs/cloud-provider-azure or https://github.com/digitalocean/digitalocean-cloud-controller-manager

Reply:
OK, those can be referred to as "cloud providers" though, right? It feels like "integration code" / "cloud provider integration code" is a more convoluted and might be difficult for users to understand when reading the code.